packages/salesforce-adapter/src/change_validators/package.ts

```diff
@@ -13,32 +13,28 @@
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
-import {
-  ActionName,
-  Change,
-  ChangeError,
-  ChangeValidator,
-  Element,
-  getAllChangeData,
-  getChangeData,
-  InstanceElement,
-  isAdditionOrRemovalChange,
-  isFieldChange,
-  isInstanceChange,
-  isModificationChange,
-  isObjectType,
-  isRemovalChange,
-} from '@salto-io/adapter-api'
+import { ChangeError, ChangeValidator, CORE_ANNOTATIONS, Element, getChangeData } from '@salto-io/adapter-api'
 import _ from 'lodash'
-import { collections } from '@salto-io/lowerdash'
-import { detailedCompare } from '@salto-io/adapter-utils'
-import { apiName, isCustomObject, metadataType } from '../transformers/transformer'
-import { INSTALLED_PACKAGE_METADATA, NAMESPACE_SEPARATOR } from '../constants'
+import { collections, values } from '@salto-io/lowerdash'
+import { apiName } from '../transformers/transformer'
+import { NAMESPACE_SEPARATOR } from '../constants'
 import { INSTANCE_SUFFIXES } from '../types'
 
 const { awu } = collections.asynciterable
+const { isDefined } = values
 
 
+const createPackageElementModificationChangeWarning = (
+  { elemID, annotations }: Element,
+  namespace: string
+): ChangeError => ({
+  elemID,
+  severity: 'Warning',
+  message: 'Modification of element from managed package may not be allowed',
+  detailedMessage: `Modification of element ${elemID.getFullName()} from managed package with namespace ${namespace} may not be allowed. `
+  + `For more information refer to ${annotations[CORE_ANNOTATIONS.SERVICE_URL]} https://developer.salesforce.com/docs/atlas.en-us.sfdx_dev.meta/sfdx_dev/packaging_packageable_components.htm`,
```

Comment:
I think this is missing a comma:
```suggestion
  + `For more information refer to ${annotations[CORE_ANNOTATIONS.SERVICE_URL]}  and https://developer.salesforce.com/docs/atlas.en-us.sfdx_dev.meta/sfdx_dev/packaging_packageable_components.htm`,
```